gms/factories/src/main/java/com/linkedin/common/factory/RestHighLevelClientFactory.java

```diff
@@ -18,27 +30,59 @@
 public class RestHighLevelClientFactory {
 
   @Value("${ELASTICSEARCH_HOST:localhost}")
-  private String elasticSearchHost;
+  private String host;
 
   @Value("${ELASTICSEARCH_PORT:9200}")
-  private Integer elasticSearchPort;
+  private Integer port;
 
   @Value("${ELASTICSEARCH_THREAD_COUNT:1}")
-  private Integer elasticSearchThreadCount;
+  private Integer threadCount;
 
   @Value("${ELASTICSEARCH_CONNECTION_REQUEST_TIMEOUT:0}")
-  private Integer elasticSearchConectionRequestTimeout;
+  private Integer connectionRequestTimeout;
```

Comment:
Since spring has dependency injection, did you want to split this out to its own factory to make this file clearer? Then you can just inject the (maybe null / optional) SSL Context into this factory.

Reply:
Since the SSL Context may need to be configured separately (ES vs neo4j), having a singleton may not work here. Any suggestion?

Reply:
idk exactly how spring works, but in offspring we have different singleton and non singleton factories.

You could try having the ES setup be singleton, but SSL not be. Then the ES setup effectively has its own SSL factory.